docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,181 @@
+
+We also will publish a list of endpoints for each level of telemetry, so the given level of telemetry could be blocked on the organization level (for example by the rule on the firewall).
+
+
+If someone found that we are publishing some problematic data we will remove them and update the aggregation process to prevent such a situation in the future.
+This NAP will be updated to reflect the current state of telemetry. 
+
+ 
+## Related Work 
+
+Total systems:
+https://plausible.io/
+https://sentry.io/
+https://opentelemetry.io/
+
+Visualizations:
+https://github.com/grafana/grafana
+  
+
+  
+## Implementation 
+
+The main thing for implementation should be the low cost of maintenance. So the solution should be as simple as possible. We could either use existing solutions on the server side or implement our own.
+
+The benefit of the existing solution is that most of the work is already done. The downside is that it may require additional cost of maintenance. This cost may be caused by many features that are not needed for napari and could increase the risk of leaking data.  As I check they are implemented in techniques that are not familiar to napari core-devs. So if there will be a decision to use them we should select an SAS solution that will be maintained by the company.
+
+
+For the current, I suggest creating a simple REST API server for collecting the data. 
+It could be a simple Python FastAPI server that will store data in the SQLite database.
+
+Data for aggregation should be extracted from the database using a script running on the same machine.
+
+The output of the aggregation script should be loaded to some existing visualization tool, like grafana.
+
+It may be nice to host them on separate servers, then even if the data presented on the dashboard will be compromised, the raw data will be not exposed to the world.
+
+Having both server and aggregation scripts in Python will reduce maintenance costs for napari core-devs.
+
+We should register the `telemetry.napari.org` domain and use it for the server. The main page contains this NAP and a link to the summary dashboard.
+
+
+The main part of the application side should be implemented in `napari-telemetry` package. 
+The package should not report in stream mode, but collect data on the disk and send it in batches. This will reduce the risk of leaking data. The package should implement a utility to allow users to preview collected data before sending it to the server.
+
+In the napari itself following changes should be implemented:
+
+1) The indicator that shows the telemetry status
+2) The dialog that asks a user if they want to enable telemetry
+3) code to check if telemetry is enabled (to not load the `napari-telemetry` package if it is disabled)
+4) code required to init `napari-telemetry` package
+
+
+## Potential problems 
+
+There is a risk that someone may try to highjack the telemetry module name to have code executed at every napari start. 
```

Comment:
Can we expand on this please? I don't understand how this would happen and how examining plugin dependencies could avoid it.

Reply:
In general, there is no need for the Python package name will match the module name `scikit-image` vs `skimage`. So some package could contain `napari_telemetry` module that during installation will overwrite/create `napari_telemetry` module. 

Simple things that could be used for hardering:

1) check if `napari-telemetry` package is installed using `importlib.metadata` before try to import
2) use sha256 from package metadata to validate file before import. 

But same problem may happen with `napari_builtins` or even someone could try to edit and glue to `napari/__init__.py`, So I'm not sure. Maybe I should remove this part?

Reply:
I don't think that's a problem. Someone could make a fake numpy...

Reply:
agreed, I think this is overkill and not really related to the telemetry package specifically.